pkg/diagnostics/networkpod/pod.go

```diff
@@ -148,6 +148,13 @@ func (d CheckPodNetwork) checkConnection(pods1, pods2 []kapi.Pod, warnMsg string
 			}
 		}
 	}
+
+	if !sameNamespace && !strings.Contains(warnMsg, "global to non-global projects") {
```

Comment:
We don't want to warn in this case because same namespace can't have global and non-global pods at the same time. May be we can make the caller pass a flag to checkConnection() to check this condition or not.